avl_tree/AvlTree.java

```diff
@@ -0,0 +1,271 @@
+import java.util.Scanner;
+import java.util.NoSuchElementException;
+import java.util.Random;
+
+public class AvlTree {
+
+	public static void main(String[] args) {
+		AVL avl = new AVL();
+		Random rand = new Random();
+		int inp = 10;
+		System.out.println(avl.isEmpty());
+		for (int i = 0; i < inp; i++) {

```

Comment:
Nothing wrong but its like a matter of `luck` that the exception will be tested, but my requested changes assure it.